scipy/optimize/_lsq/givens_elimination.pyx

```diff
@@ -44,7 +47,7 @@ def givens_elimination(double [:, ::1] S, double [:] v, double [:] diag):
         if diag[i] == 0:
             continue
 
-        diag_row[:] = 0
+        diag_row[i+1:] = 0
```

Comment:
> >    The new and old code are only equivalent is diag_row[i] really is reset to 0

> Can you explain why it seems to you like this? Maybe I'm missing something.

Probably because I'm reading too fast, but the current code does:

    diag_row[:] = 0
    diag_row[i] = diag[i]

in each iteration, so will contain ``[0, 0, 0, 0.23432, 0, 0, 0]`` (for ``i==3``). The new code only resets elements ``i+1:``, so (for ``n==4``) could contain ``[0.11, 0.22, 0.33, 0.23432, 0, 0, 0]``.

The above isn't true if the comment about implicit 0 really resets element ``j`` to identically 0, which I didn't verify.